### ISSUE TITLE
Rollup is significantly slower when run simultaneously in different processes

### ISSUE THREAD

**Issue by getkey** [rollup/rollup#896]
I need to create 3 bundles. When I create them in a single process, it takes ~8 seconds.

But since those 3 tasks are independent, I thought running them concurrently in separate processes (with the `cluster` module) would make bundling  them faster. But it actually takes ~16 seconds!

Given a project containing the following files:

``` JavaScript
// app.js
import { count, increment } from './counter';

console.log(count); // 0
increment();
console.log(count); // 1
```

``` JavaScript
// counter.js
export var count = 0;
export function increment () {
  count += 1;
}
```

A traditional bundler takes 0.17 seconds to be run:

``` JavaScript
// bundler.js
const rollup = require('rollup');

function errorHandler(err) {
    console.error(err);
}
for (let oPath of ['./bundle1.js', './bundle2.js', 'bundle3.js']) {
    rollup.rollup({
        entry: './app.js',
    }).then((bundle) => {
        return bundle.write({
            format: 'cjs',
            exports: 'none',
            indent: false,
            sourceMap: 'inline',
            cache: oPath,
            dest: oPath
        });
    }).catch(errorHandler);
}
```

``` shell
> time node bunler.js 
0.17user 0.02system 0:00.20elapsed 98%CPU (0avgtext+0avgdata 29436maxresident)k
0inputs+24outputs (0major+4963minor)pagefaults 0swaps
```

A parallelized bundler takes 0.71 seconds.

``` JavaScript
// bundler_cluster.js
const cluster = require('cluster'),
    rollup = require('rollup');

function errorHandler(err) {
    console.error(err);
}

if (cluster.isMaster) {
    for (let oPath of ['./bundle1.js', './bundle2.js', 'bundle3.js']) {
        let worker = cluster.fork();
        worker.on('online', () => {
            worker.send(oPath);
        });
        worker.on('exit', (code, signal) => {
            if (code !== 0) console.err('Process exited eroneously with code: ' + code + ', signal: ' + signal);
        });
    }
} else if (cluster.isWorker) {
    process.on('message', (oPath) => {
        rollup.rollup({
            entry: './app.js',
        }).then((bundle) => {
            return bundle.write({
                format: 'cjs',
                exports: 'none',
                indent: false,
                sourceMap: 'inline',
                cache: oPath,
                dest: oPath
            });
        }).catch(errorHandler
        ).then(() => {
            process.exit(0);
        });
    });
}
```

``` shell
> time node bunler_cluster.js 
0.71user 0.05system 0:00.52elapsed 147%CPU (0avgtext+0avgdata 29836maxresident)k
0inputs+24outputs (0major+18712minor)pagefaults 0swaps
```

Why is that?
If needed I can of course provide both versions of the bundler from my actual project.

**Comment by getkey** [comment]
I obtain the same results when I have one file for each task and use the shell to make these tasks concurrent.
```sh
#! /bin/sh
node build/b1.js &
node build/b2.js &
node build/b3.js &
wait
```

I don't think I/O is the limiting factor since I get the same results on SSDs and HDDs (I've tried two of each).

**Comment by wagerfield** [comment]
Have any investigations been done into this? Is Rollup able to utilise multithreading for faster bundle times?

**Comment by guybedford** [comment]
Closing as this has been inactive for a while and newer relevant issues can track these concerns. Happy to continue discussion as well here too.

**Comment by derrickb** [comment]
@guybedford can you share what issues we should discuss this in? I did a search for 'parallel' and 'concurrent' and didn't find anything.

I found this via google after attempting to make my grunt tasks parallel with `grunt-concurrent`. I am generating two bundles, and am also finding it to be slower (by at least 2x) when ran in parallel.

**Comment by mreinstein** [comment]
so the resolution is no one has solved it or looked at it, so it's resolved?

*scratches head*

**Comment by shellscape** [comment]
@mreinstein You're replying to an issue almost two years old. the number of versions released and changes since then are myriad. If this is still affecting you, please open a new issue. That follows Guy's reply before closing.

You'll find that replying to very old issues like this one frowned upon generally on GitHub.
